workflows/wf_theiacov_illumina_pe.wdl

```diff
@@ -38,92 +45,130 @@ workflow theiacov_illumina_pe {
+        reference_genome = reference_genome,
+        consensus_min_depth = min_depth
     }
-    call taxon_ID.nextclade_output_parser_one_sample {
+    call consensus_qc_task.consensus_qc as consensus_qc {
+        input:
+          assembly_fasta =  consensus.consensus_seq,
+          reference_genome = reference_genome,
+          genome_length = genome_length
+      }
+    call assembly_metrics.stats_n_coverage {
       input:
-      nextclade_tsv = nextclade_one_sample.nextclade_tsv
+        samplename = samplename,
+        bamfile = bwa.sorted_bam
+    }
+    if (organism == "sars-cov-2") {
+      # sars-cov-2 specific tasks
+      call taxon_ID.pangolin4 {
+        input:
+          samplename = samplename,
+          fasta = consensus.consensus_seq
+      }
+      call sc2_calculation.sc2_gene_coverage {
+        input: 
+          samplename = samplename,
+          bamfile = bwa.sorted_bam,
+          min_depth = min_depth
+      }
+    }
+    if (organism == "MPXV" || organism == "sars-cov-2"){ 
+      # tasks specific to either MPXV or sars-cov-2
+      call taxon_ID.nextclade_one_sample {
+        input:
+          genome_fasta = consensus.consensus_seq,
+          dataset_name = select_first([nextclade_dataset_name, organism]),
+          dataset_reference = nextclade_dataset_reference,
+          dataset_tag = nextclade_dataset_tag
+      }
+      call taxon_ID.nextclade_output_parser_one_sample {
+        input:
+        nextclade_tsv = nextclade_one_sample.nextclade_tsv
+      }
     }
+    if (organism == "MPXV" || organism == "sars-cov-2" || organism == "WNV"){ 
+      # tasks specific to MPXV, sars-cov-2, and WNV
+      call ncbi.vadr {
+          input:
+            genome_fasta = consensus.consensus_seq,
+            assembly_length_unambiguous = consensus_qc.number_ATCG
+        }
+      }
+    String assembly_method_nonflu = "~{bwa.bwa_version}; ~{primer_trim.ivar_version}"
   }
-  if (organism == "MPXV" || organism == "sars-cov-2" || organism == "WNV"){ 
-    # tasks specific to MPXV, sars-cov-2, and WNV
-    call ncbi.vadr {
+  if (organism == "flu"){
+    # flu-specific tasks
+    call irma.irma {
       input:
-        genome_fasta = consensus.consensus_seq,
-        assembly_length_unambiguous = consensus_qc.number_ATCG
+        read1 = read_QC_trim.read1_clean,
+        read2 = read_QC_trim.read2_clean,
+        samplename = samplename,
+    }
+    if (defined(irma.irma_assemblies)) {
+      call abricate_flu.abricate_flu {
+        input:
+          assembly = select_first([irma.irma_assembly_fasta]),
+          samplename = samplename,
+          nextclade_flu_h1n1_tag = nextclade_flu_h1n1_tag,
+          nextclade_flu_h3n2_tag = nextclade_flu_h3n2_tag,
+          nextclade_flu_vic_tag = nextclade_flu_vic_tag,
+          nextclade_flu_yam_tag = nextclade_flu_yam_tag,
+      }
+      call consensus_qc_task.consensus_qc as consensus_qc_flu {
```

Comment:
is there a reason we are having these run as separate tasks? I think we could run consensus_qc for all organisms instead of having the task show up twice?